libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,314 @@
+package protocols
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+)
+
+type Address string
+type Performative []string
+type SomeMessageType string
+
+const (
+	NONCE_BYTES_NB = 32
+)
+
+type Role string
+
+const (
+	Role1             Role = "role1"
+	Role2             Role = "role2"
+	StartingMessageId      = 1
+	StartingTarget         = 0
+)
+
+type RuleType struct {
+	initialPerformative  Performative
+	terminalPerformative Performative
```

Comment:
Not sure if these are singular or can be a collection later on, but initialPerformative and terminalPerformative are sets of performatives. So type-wise ensure that is possible.
Also then name-wise, initialPerformatives and terminalPerformatives (with s at the end).